### ISSUE TITLE
Appositives for noun phrase post modifier

### ISSUE THREAD

**Issue by AlJohri** [simplenlg/simplenlg#35]
Hi, I'm having an issue similar to @jeremybmerrill with appositives in issue #15.

I'm trying to create an appositive for the subject of my sentence and I would like the commas to appear on either side of the post modifier.

Using a simplenlg jruby wrapper, my code looks like the following:

```
subject = NLG.factory.create_noun_phrase('Dude')
modifier = NLG.factory.create_noun_phrase('a Republican')
modifier.set_feature(NLG::Feature::APPOSITIVE, true)
subject.add_post_modifier(modifier)

verb = NLG.factory.create_verb_phrase('face')
verb.set_feature(NLG::Feature::PROGRESSIVE, true)

object = NLG.factory.create_noun_phrase('Dudette')

clause = NLG.factory.create_clause(subject, verb, object)
puts NLG.realizer.realise_sentence(clause)
```

This renders as:

>Dude, a Republican is facing Dudette.

where I would expect it to render as:

>Dude, a Republican, is facing Dudette.

I've only seen the comma appear properly when adding prepositional phrases to the clause itself similar to the test cases here: https://github.com/simplenlg/simplenlg/blob/a9995a69ae15e2ca8ba02b82a6cf2c8367e7728a/src/test/java/simplenlg/syntax/english/OrthographyFormatTest.java

Am I using this feature incorrectly? I felt like a prepositional phrase was not the right thing to use in this situation.

Thanks!

**Comment by dmil** [comment]
+1

**Comment by jeremybmerrill** [comment]
Ha, it's like a NICAR convention in here all of a sudden! 

I'd be happy to help you guys out in a few days when I get my dev environment set up on my new computer...

**Comment by jeremybmerrill** [comment]
Hey @aljohri, @dmil... try out the master branch on my fork. https://github.com/jeremybmerrill/simplenlg. Commit: https://github.com/jeremybmerrill/simplenlg/commit/9ebb9ae7d327e94aa54617b0c34a56bd8ada5607

I made it so that, for post-modifier appositives, there's a final comma.

I made a change to a test -- where the expected result seemed incorrect to me. (It was "The dog, a rottweiler, runs." now it's "The dog, a rottweiler, runs.".) *Do y'all agree with that judgment?*

All tests pass. I can put in a PR in a few days..., @Saad-Mahamood, if that looks good to you.
